### PR TITLE
Upgrade BookKeeper to 4.10.0

### PR DESCRIPTION
**Pull request by eolivelli**
Upgrade BookKeeper and fix our custom EnsamblePlacemenetPolicy

### CODE REVIEW

**Comment by mino181295** [comment]
+1 

Travis is stalling on _ScanHugeTableTest_ maybe re-run the CI

**Comment by eolivelli** [comment]
During the update of Pulsar we saw that there is a new variant of NoSuchLedgerException (NoSuchLedgerExistsOnMetadataServerException), we should deal with it, it is strange that tests are passing
https://github.com/apache/pulsar/pull/5607

File: herddb-core/src/main/java/herddb/cluster/PreferLocalBookiePlacementPolicy.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ public void registerSlowBookie(BookieSocketAddress bookieSocketAddress, long ent
             newBookies.add(bookie);
             --ensembleSize;
             if (ensembleSize == 0) {
-                return PlacementResult.of(newBookies, false);
+                return PlacementResult.of(newBookies, PlacementPolicyAdherence.MEETS_SOFT);
```
[the remark this answers is not in the document]
Comment (eolivelli) [reply herddb-core/src/main/java/herddb/cluster/PreferLocalBookiePlacementPolicy.java]: @amitvc 
thanks for pointing this out

It has been implemented in https://github.com/apache/bookkeeper/commit/b34162a932b93a7e65451dce3f6e10bfda707997

Honestly for us there is no much difference between MEETS_SOFT and MEETS_STRICT.
So I am changing it to MEETS_STRICT as is it in the Default policy of BookKeeper

**Comment by eolivelli** [comment]
@amitvc @diegosalvi  I have added a new commit, PTAL

**Comment by amitvc** [comment]
@eolivelli LGTM

**Comment by eolivelli** [comment]
Thanks.
We can merge this patch and so the same in blobit

**Comment by eolivelli** [comment]
Please @amitvc click 'approve'

**Comment by amitvc** [comment]
@eolivelli I don't see an option to approve. I was going to write you an email to report the same.

**Comment by eolivelli** [comment]
It should be in the third tab, 'changes'

**Comment by eolivelli** [comment]
I can't merge without at least one approval

**Comment by amitvc** [comment]
Done.

**Pull request merged**